aiohttp/client.py

```diff
@@ -295,12 +303,22 @@ def request(self,
             for i in skip_auto_headers:
                 skip_headers.add(istr(i))
 
-        if proxy is not None:
+        if proxy is not None and self._session_proxy is None:
             try:
                 proxy = URL(proxy)
             except ValueError:
                 raise InvalidURL(proxy)
 
+        if proxy is None and self._session_proxy is not None:
+            try:
+                proxy = self._session_proxy
+                proxy_auth = self._session_proxy_auth
```

Comment:
By this line you are overriding explicitly passed `proxy_auth` and `proxy_headers` arguments with session default ones.

I have no idea how to resolve the conflict properly but proposed behavior looks confusing.

Reply:
I am trying to use session proxy if proxy is not mentioned explicitly for a particular request.